src/bap_mc/bap_mc.ml

```diff
@@ -15,10 +15,10 @@ exception Unknown_arch
 exception Can't_lift of Error.t
 exception Trailing_data of int
 
-let no_disassembly state boff =
+let no_disassembly state boff offset =
   let mem = Dis.memory state in
   let addr = Dis.addr state in
-  let stop = Addr.to_int addr |> ok_exn in
+  let stop = (Addr.to_int addr |> ok_exn) - offset in
```

Comment:
the arithmetics should be done in `Addr` algebra, not in `Int`, otherwise all negatives would be incorrect. With this said, `offset` is a bad name - as it is not an offset in the data, it is starting address of the first byte.